libavfilter/dnn_filter_common.c

```diff
@@ -19,6 +19,44 @@
 #include "dnn_filter_common.h"
 #include "libavutil/avstring.h"
 
+static int ff_dnn_check_output_nb(AVFilterContext *ctx, DNNFunctionType func_type, DNNBackendType backend_type, int output_nb)
+{
+    switch(func_type) {
+    case DFT_PROCESS_FRAME:
+        if (output_nb != 1) {
+            av_log(ctx, AV_LOG_ERROR, "Dnn processing filter needs 1 output only, \
+                                       but get %d instead\n", output_nb);
+            return AVERROR(EINVAL);
+        }
+        return 0;
+    case DFT_ANALYTICS_DETECT:
+        switch(backend_type) {
+        case DNN_TF:
+            if (output_nb != 4) {
```

Comment:
this logic is special in vf_detect.c for tf backend, why we check here?

Reply:
I think maybe tf classify model may reuse this logic code in following code. Since tf calssify model needs multiple outputs as well.

Reply:
the magic number 4 should be special

Reply:
OK~ So is it ok that i put the logic here?

Reply:
just put in vf_dnn_detect.c now, we can refine the code if it is common later

Reply:
OK, got you